src/main/java/ui/components/pickers/MilestonePickerDialog.java

```diff
@@ -43,6 +43,20 @@ public MilestonePickerDialog(Stage stage, TurboIssue issue, List<TurboMilestone>
         state = new MilestonePickerState(originalMilestones);
         initUI();
         setupKeyEvents();
+        fillInputField();
+    }
+
+    /**
+     * Fills the input field of milestone picker dialog
+     * If there is an existing milestone, input field is filled with existing milestone's title
+     * Else input field is filled with the next open milestone that is not overdue
+     */
+    private void fillInputField() {
+        if (!getExistingMilestone(originalMilestones).isPresent()) {
+            getNextOpenMilestone(originalMilestones)
+                    .map(PickerMilestone::getTitle)
+                    .ifPresent(this::fillInputFieldWithMilestoneName);
+        }
```

Comment:
This will call `getExistingMilestone` twice (here and below). How about something like
```
// Fill X with the relevant values
Optional<X> defaultMilestone = getExistingMilestone(X).orElse(getNextOpenMilestone(X)).

defaulMilestone.map(X).ifPresent(this::fillInputFieldWithMilestoneName);
```